types/react/CHANGELOG.md

```diff
@@ -0,0 +1,316 @@
+| `WeakValidationMap`     | [`deprecated-prop-types-types`](https://github.com/eps1lon/types-react-codemod#deprecated-prop-types-tpyes)               | `WeakValidationMap` from `prop-types`    |
+
+#### Not Codemoddable
+
+During our [example migrations](https://github.com/users/eps1lon/projects/3/views/9), these types were not used at all.
+So, we felt comfortable leaving them off our codemod list.
+If you feel a codemod is missing, it can be tracked in the [list of missing React 19 codemods](https://github.com/eps1lon/types-react-codemod/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22React+19%22+label%3Aenhancement).
+
+| Type                    | Replacement                                                                                                    |
+| ----------------------- | -------------------------------------------------------------------------------------------------------------- |
+| `ClassicComponentClass` | `ClassicComponentClass` from `create-react-class`                                                              |
+| `ClassicComponent`      | `ClassicComponent` from `create-react-class`                                                                   |
+| `ClassicElement<Props>` | `ClassicElement<Props, InstanceType<T>>` from `create-react-class`                                             |
+| `ComponentSpec`         | `ComponentSpec` from the `create-react-class` package                                                          |
+| `Mixin`                 | `Mixin` from the `create-react-class` package                                                                  |
+| `ReactChildren`         | `typeof React.Children`                                                                                        |
+| `ReactHTML`             | Either `ReactHTML` from `react-dom-factories` or, if you used `keyof ReactHTML`, use `HTMLElementType` instead |
+| `ReactSVG`              | Either `ReactSVG` from `react-dom-factories` or, if you used `keyof ReactSVG`, use `SVGElementType` instead    |
+| `SFCFactory`            | No replacement                                                                                                 |
+
+### JSX Namespace
+
+A long-time request is to remove the global `JSX` namespace from our types in favor of `React.JSX`.
+This helps prevent pollution of global types which prevents conflicts between different UI libraries that leverage JSX.
+This change is [codemoddable with `scoped-jsx`](https://github.com/eps1lon/types-react-codemod#scoped-jsx).
+
+This means that if you're doing any module augmentation of `JSX`:
+
+```ts
+// global.d.ts
+
+// This adds a new global JSX element
+namespace JSX {
+  interface IntrinsicElements {
+    "my-element": {
+      myElementProps: string;
+    };
+  }
+}
+```
+
+You'll now need to wrap it in `declare module "...."`:
+
+```diff
+// global.d.ts
+
++ declare module "react" {
```

Comment:
I think `namespace React` being available is some old legacy behavior of TypeScript that you shouldn't really rely on. Always use `import * as React` instead of assuming TypeScript makes it available without import.

It would break anyway as soon as we move to ESM so let's not bless this pattern.

Reply:
We'd also have to double check if it works to begin with. I haven't used `namespace React` in ages for module augmentation so I can't say if this works.

Reply:
It definitely works - the React types makes the namespace globally available.

Reply:
Hm, need to figure out if we do this accidentally or TypeScript is doing that intentionally. Either way, the `declare module` is the way to go not some implicit, globally available namespace.

Reply:
It's intentional that it's global. But relying on it is another breaking change you have to resolve when React moves to ES modules and types will so as well. Better to start relying on `declare module 'react'` instead of `declare namespace React`